### PR TITLE
Fix KafkaConnectIntegrationTest.test_reading_and_writing_to_map test  [HZ-3116]

### PR DESCRIPTION
**Pull request by orcunc**
The test is downloading the DataGen connector zip file from JFrog. In one of the test runs it failed with error message :
 java.net.SocketException: Connection reset

 It would be better to have the connector zip as local resource to avoid download failures in the future  [HZ-3116]

Fixes : https://github.com/hazelcast/hazelcast/issues/24905

Breaking changes (list specific methods/types/messages):
* API
* client protocol format
* serialized form
* snapshot format

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

[HZ-3116]: https://hazelcast.atlassian.net/browse/HZ-3116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by TomaszGaweda — CHANGES_REQUESTED** [review]
I don't really like copying dependencies into the project.

In hazelcast3 connector we had a trick, that we used Maven to download the jars and then use it locally. We can do the same for KafkaConnect ITs, just:
- make it so that it's automated
- implement for all tests

**Comment by orcunc** [comment]
The confluentinc-kafka-connect-datagen is not available on maven central. But we have in on JFrog. download-maven-plugin is going to download and cache it at ${local-repo}/.cache/maven-download-plugin

Documentation is : https://maven-download-plugin.github.io/maven-download-plugin/docsite/1.7.1/wget-mojo.html

File: extensions/kafka-connect/pom.xml [hunk 1]
```diff
@@ -32,6 +32,14 @@
         <version>5.4.0-SNAPSHOT</version>
     </parent>
 
+    <!-- Repository to download kafka-connect-datagen. Because this resource does not exist on maven central    -->
+    <repositories>
```
Comment (ldziedziul) [inline extensions/kafka-connect/pom.xml:36]: Do we need that?

File: extensions/kafka-connect/pom.xml [hunk 2]
```diff
@@ -118,6 +146,13 @@
             <version>4.4.12</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (ldziedziul) [inline extensions/kafka-connect/pom.xml:149]: Do we need it?

**Review by TomaszGaweda — APPROVED** [review]

**Review by ldziedziul — APPROVED** [review]

**Pull request merged**